### DISCUSSION TITLE
Keeping song album in compilation

### DISCUSSION THREAD

**Discussion by Serene-Arc** [beetbox/beets#4481] [Q&A] [answered]
Hi! I've got a question with regards to importing compilations. Most of the compilations I'm importing don't have any matches on MusicBrainz so the autotagger is turned off when importing.

When I import a compilation, the first track in the compilation names the compilation folder that the actual files are stored in. When I load the files into Audacious at this stage, all of the individual songs have their original album name in the metadata. However, when I use `modify` or `edit` to change the album name of the compilation, that change propgates to all of the individual songs, so their album becomes the name of the compilation.

What I'm wondering is this: is there a way to retitle the compilation *without* changing the album of the tracks? I'd like to preserve as much of the original metadata as possible.

**Comment by sampsyo** [comment, marked as answer]
That's a tricky one—it's an assumption in beets that there are certain _album-level fields_ that are nearly always consistent across all the tracks in a given album. It's possible to violate this temporarily, but commands like `modify` and `edit` will always do their best to keep those fields consistent.

One option you might consider is to use a _different_ field for the non-compilation album that the track is part of. For example, invent a field called `orig_album` or something like that and put the "original" album that the track was taken from in there. Then `album` would be the name of the compilation. Would that make any sense for your use case?

**Reply by Serene-Arc** [reply]
It would, but I was wondering if there was a mechanism to do it in beets. Basically I want to use the compilation name as a group and wrapper for the album, and beets does seem to do this, if only unintentionally. When I import the compilation, all of the tracks can be referred to by the album name of the first track; it's just that I want to change that name and location on disk. Whether this is a bug or unintentioned, I don't know.

Is there any easy way to set a field like that in bulk for a group of tracks?

EDIT: Never mind, I upgraded to the latest release and there's an option to use templates with the `modify` command! Excellent, thanks.
